### PR TITLE
replace createAppContainer

### LINKED ISSUE #2537: Replace createAppContainer
## Describe the bug

The method `createAppContainer` has been removed from react-navigation v5
Currently if you try to dispense in the app, the `TabNavigator` throws an error

### To reproduce

1. Go to Dispensing
2. Dispense

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #2537 

## Change summary

Replaces the call to `createAppContainer` with the declarative `Tab.Navigator`
This is working for the tabbing - but there is a subsequent problem with the `PrescriptionPage` which is now not receiving the `transaction` prop.

Have also modified the shape of the `tabs` supplied to the `TabNavigator` to bring in the title and pass in a name which is required by the `Tab.Screen` component.

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] Run through the dispensing wizard. Should all function correctly.

### Related areas to think about

No, I think that's about it.

### CODE REVIEW

**Comment by joshxg** [comment]
Oh, the background of all the tabs is now gray rather than the blueish gray it was previously

File: src/widgets/TabNavigator.js [hunk 1]
```diff
@@ -23,26 +16,25 @@ const DEFAULT_TAB_CONFIG = { navigationOptions: { tabBarVisible: false } };
...
-  const Container = useMemo(() => {
-    const tabsConfig = tabs.reduce(
-      (acc, value, index) => ({ ...acc, [index]: { ...DEFAULT_TAB_CONFIG, screen: value } }),
-      {}
-    );
-    const defaultNavigationConfig = { initialRouteName: String(currentTabIndex) };
-    return getTabNavigator(tabsConfig, defaultNavigationConfig);
-  }, []);
+  const navigatorRef = useRef(useNavigation());
+  const Tab = createBottomTabNavigator();
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply src/widgets/TabNavigator.js]: 782ab378

Actually, it made no difference. I tested by creating a new Screen component which generated a random number on render and there was no re-render either way.

File: src/widgets/Wizard.js [hunk 2]
```diff
@@ -20,14 +20,14 @@ import { BACKGROUND_COLOR, BLUE_WHITE, SHADOW_BORDER } from '../globalStyles/col
...
-const WizardComponent = ({ tabs, titles, currentTab, switchTab }) => (
+const WizardComponent = ({ tabs, currentTab, switchTab }) => (
   <View style={localStyles.container}>
     <View style={localStyles.stepperContainer}>
       <Stepper
         numberOfSteps={tabs.length}
         currentStep={currentTab}
         onPress={switchTab}
-        titles={titles}
+        titles={tabs.map(tab => tab.title)}
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply src/widgets/Wizard.js]: 6b28724b

File: src/widgets/TabNavigator.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshxg) [reply src/widgets/TabNavigator.js]: uhh.. 



embares

**Comment by mark-prins** [comment]
> I see there's some other issue for the params so will approve this one !

Oops! Sorry @joshxg - I fixed the background colour ( 234275bb ) and in doing that accidentally discarded your review.

**Comment by mark-prins** [comment]
@joshxg Yes - had noticed that. was thinking of creating a new issue for it; will keep this one alive for a bit longer while I try to figure out the neatest way

**Comment by Rebecca367** [comment]
Have done some testing, mobile is able to get info. from dispensary module now. 
Well done guys!
- [x] Run through the dispensing wizard. Should all function correctly.

**Pull request merged**